### PR TITLE
Use a random available port for the mock DNS server in tests

### PR DESCRIPTION
**Pull request by Cnly**
Travis CI builds have failed toooooo many times.

Signed-off-by: Alex Chen <minecnly@gmail.com>

### CODE REVIEW

File: resolve_test.go [hunk 1]
```diff
@@ -236,22 +232,33 @@ func TestResolutionHostnameWithNoWellKnownNorSRV(t *testing.T) {
...
 	go func() {
-		err := srv.ListenAndServe()
+		err := srv.ActivateAndServe()
 		if err != nil {
 			panic(err)
 		}
 	}()
 
 	// Redefine the default resolver so it uses our local server.
+	actualListenAddr := udpConn.LocalAddr().String()
```
Comment (anoadragon453) [inline resolve_test.go]: A comment for why this is necessary would be nice.

Comment (Cnly) [inline resolve_test.go]:
```suggestion
```

File: resolve_test.go [hunk 2]
```diff
@@ -236,22 +232,33 @@ func TestResolutionHostnameWithNoWellKnownNorSRV(t *testing.T) {
...
+	// Start a DNS server with our custom handler on a random available port.
+	listenAddr, err := net.ResolveUDPAddr("udp", "127.0.0.1:0")
+	if err != nil {
+		panic(fmt.Sprintf("failed to ResolveUDPAddr: %v", err))
+	}
+	udpConn, err := net.ListenUDP("udp", listenAddr)
+	if err != nil {
+		panic(fmt.Sprintf("failed to ListenUDP: %v", err))
+	}
+
```
Comment (Cnly) [inline resolve_test.go]:
```suggestion

	// Get the actual address and port we're listening on
	actualListenAddr := udpConn.LocalAddr().String()

```

File: resolve_test.go (hunk 1 above)
Comment (Cnly) [reply resolve_test.go]: Does it look better now?

Comment (anoadragon453) [reply resolve_test.go]: Yep!

**Pull request merged**